TransformAST.cpp

```diff
@@ -0,0 +1,142 @@
+            
+            swift::ParameterList *empty_params_list =
+                swift::ParameterList::createEmpty(ast_ctx);
+
+            llvm::StringRef fn_name_str = src_file.getFilename();
+            swift::Identifier fn_name_id = ast_ctx.getIdentifier(fn_name_str);
+            swift::DeclName fn_name(ast_ctx, fn_name_id, empty_params_list);
+            
+            swift::BraceStmt *old_top_level_body = last_top_level_code_decl->getBody();
+            swift::SourceLoc fn_start = old_top_level_body->getLBraceLoc();
+            swift::SourceLoc fn_end = old_top_level_body->getRBraceLoc();
+            
+            swift::FuncDecl *new_func = swift::FuncDecl::create(ast_ctx, // AST Context
+                                                                swift::SourceLoc(), swift::StaticSpellingKind::None, // StaticLoc, StaticSpelling
+                                                                fn_start, fn_name, fn_start, // FuncLoc, Name, NameLoc
+                                                                false, swift::SourceLoc(),   // Throws, ThrowsLoc
+                                                                nullptr, // GenericParams
+                                                                empty_params_list, // BodyParams
+                                                                swift::TypeLoc(), // ReturnType should be void
+                                                                last_top_level_code_decl); // Parent
+            swift::BraceStmt *fn_body = swift::BraceStmt::create(ast_ctx,
+                                                                 swift::SourceLoc(),
+                                                                 old_top_level_body->getElements(),
+                                                                 swift::SourceLoc(),
+                                                                 true);
+            new_func->setBody(fn_body);
+            
+            swift::BraceStmt *new_top_level_body = swift::BraceStmt::create(ast_ctx,
+                                                                            fn_start,
+                                                                            llvm::ArrayRef<swift::ASTNode>(new_func),
+                                                                            fn_end,
+                                                                            true);
+            last_top_level_code_decl->setBody(new_top_level_body);
+        }
+    }
+}
+
+void MakeDeclarationsPublic(swift::SourceFile &src_file)
+{
+    class Walker : public swift::ASTWalker
+    {
+        bool CanBeMadePublic(swift::Decl *decl)
+        {
+            if(llvm::isa<swift::StructDecl>(decl->getDeclContext()))
+            {
+                if(auto var = llvm::dyn_cast<swift::VarDecl>(decl))
+                    if(var->getOriginalWrappedProperty())
+                        return false;
+                return true;
```

Comment:
You could make the condition after this (cast to AccessorDecl) an `else if`, and avoid returning true here (as in, just let it fall through)